retis/src/core/filters/meta/filter.rs

```diff
@@ -379,8 +394,15 @@ impl Rval {
 pub(crate) struct FilterMeta(pub(crate) Vec<MetaOp>);
 
 impl FilterMeta {
-    fn check_one_walkable(t: &Type, ind: &mut u8) -> Result<bool> {
+    fn check_one_walkable(t: &Type, ind: &mut u8, casted: bool) -> Result<bool> {
         match t {
+            Type::Int(i) => {
+                if casted && i.size() == 8 {
+                    *ind += 1;
+                } else {
+                    bail!("found {} while walking types", t.name())
+                }
+            }
```

Comment:
although we don't support 32bits, I think this could be converted into something like [sizeof(void *)](https://godbolt.org/z/objMxvzG9)

In that case `>=` would make sense but honestly I would not add that support (e.g. for cases like `(struct foo *)(num >> 4)`)